src/scripts/main.js

```diff
@@ -1,3 +1,20 @@
 'use strict';
 
 // write code here
+const list = document.querySelectorAll('li');
+const listArray = Array.from(list);
+
+const sortedList = listArray.sort((a, b) => {
+  const salaryA = parseInt(a.dataset.salary.replace(/[$,]/g, ''), 10);
+  const salaryB = parseInt(b.dataset.salary.replace(/[$,]/g, ''), 10);
+
+  return salaryB - salaryA;
+});
```

Comment:
The sorting logic should be encapsulated within a `sortList` function as per the task requirements. Consider defining a function `sortList` that takes the list as an argument and returns the sorted list.